src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.WithFrozenSourceGeneratedDocumentsCompilationTracker.cs

```diff
@@ -145,28 +143,6 @@ public Task<VersionStamp> GetDependentVersionAsync(SolutionCompilationState comp
         public Task<VersionStamp> GetDependentSemanticVersionAsync(SolutionCompilationState compilationState, CancellationToken cancellationToken)
             => UnderlyingTracker.GetDependentSemanticVersionAsync(compilationState, cancellationToken);
 
-        public Task<Checksum> GetDependentChecksumAsync(SolutionCompilationState compilationState, CancellationToken cancellationToken)
-        {
-            if (_lazyDependentChecksum == null)
-            {
-                var tmp = compilationState; // temp. local to avoid a closure allocation for the fast path
-                // note: solution is captured here, but it will go away once GetValueAsync executes.
-                Interlocked.CompareExchange(
-                    ref _lazyDependentChecksum,
-                    AsyncLazy.Create(static (arg, c) =>
-                        arg.self.ComputeDependentChecksumAsync(arg.tmp, c),
-                        arg: (self: this, tmp)),
-                    null);
-            }
-
-            return _lazyDependentChecksum.GetValueAsync(cancellationToken);
-        }
-
-        private async Task<Checksum> ComputeDependentChecksumAsync(SolutionCompilationState compilationState, CancellationToken cancellationToken)
```

Comment:
>ComputeDependentChecksumAsync [](http://example.com/codeflow?start=37&length=29)

Am I reading this right in that the old code would potentially reuse the cached checksum from the underlying tracker and only need to get new checksum information from _replacementDocumentStates?

Reply:
Correct. But I don't think the complexity is worth it. Having the new system operate uniformly on any Project instance is more desirable for me.